### ISSUE TITLE
🐛 Build broken due to duplicate registered views for GooglePayButton and ApplePayButton

### ISSUE THREAD

**Issue by walkerdb** [stripe/stripe-react-native#1225]
## **Describe the bug**
We're trying out `v0.22.0` to try the new `createPlatformPayPaymentMethod` flow. With the new version our metro builds fail with the following:

```
[Sat Dec 03 2022 14:23:46.213]  ERROR    Invariant Violation: Tried to register two views with the same name GooglePayButton
```

This seems to be due to the fact that `@stripe/stripe-react-native` is now internally calling 

```ts
requireNativeComponent<any>('GooglePayButton')
```

in two different places, once in `src/components/GooglePayButton.tsx` and once in `src/components/PlatformPayButton.tsx`.  If we comment out one of the requires (in our case `src/components/GooglePayButton.tsx`), the build works again*

_* after resolving the GooglePayButton issue we saw the same thing for ApplePayButton, resolved in the same way._

## To Reproduce
Run `yarn start` for your app after importing `createPlatformPayPaymentMethod`, then start up an ios simulator to start the metro dev-server compilation process.

## Expected behavior
We can build our app with the new `@stripe/stripe-react-native` version.

## Additional context
We're on `react-native@63`, which may be a factor if you're unable to reproduce on your own machine (further bumping to latest react-native for us was blocked by `tipsi-stripe` issues, which is why we're trying to migrate to `@stripe/stripe-react-native`).

**Comment by OooCleMooO** [comment]
+1

**Comment by charliecruzan-stripe** [comment]
So I'm only able to reproduce this when importing and _rendering_ both components. Not ideal, but I wouldn't really recommend using both those APIs at once anyways. Let me know if there's a use case for that though.

If this is happening _without rendering both components_, then maybe this has to do with your RN version- I tested this down to RN 0.64 (which is the lowest RN version supported as of v0.8.0 of `stripe-react-native`).

**Comment by walkerdb** [comment]
hmmm we're not importing those  components though. Here's an exhaustive list of imports that we're using:

```
import { isApplePaySupported, Token, CardField, createToken, createPlatformPayPaymentMethod, StripeProvider } from '@stripe/stripe-react-native';
import { CardFieldInput } from '@stripe/stripe-react-native/src/types`;
import { CartSummaryItem } from '@stripe/stripe-react-native/src/types/PlatformPay';
```

I guess I wouldn't be surprised if the latter two imports had resolution side-effects that end up including something from the various Button components? But we currently need them since we need a type for the `onCardChange` callback param (`CardFieldInput.Details`) and for `cartItem` items.

**Comment by charliecruzan-stripe** [comment]
Hm, copy/pasting those imports still doesn't trigger the error for me on RN 64

**Comment by walkerdb** [comment]
hmm ok. We'll move forward with a [patch-package](https://github.com/ds300/patch-package) approach then until we can update our react-native version.

If there's anyone else on < react-native 0.64, here's our patch file. If you use it you can't directly use `ApplePayButton` or `GooglePayButton`, but you should be using `PlatformPayButton` instead regardless.

```diff
diff --git a/node_modules/@stripe/stripe-react-native/src/components/ApplePayButton.tsx b/node_modules/@stripe/stripe-react-native/src/components/ApplePayButton.tsx
index 9be434f..34aec20 100644
--- a/node_modules/@stripe/stripe-react-native/src/components/ApplePayButton.tsx
+++ b/node_modules/@stripe/stripe-react-native/src/components/ApplePayButton.tsx
@@ -7,8 +7,8 @@ import {
   ViewStyle,
 } from 'react-native';
 
-const ApplePayButtonNative =
-  requireNativeComponent<ApplePayButtonComponent.NativeProps>('ApplePayButton');
+// const ApplePayButtonNative =
+//   requireNativeComponent<ApplePayButtonComponent.NativeProps>('ApplePayButton');
 
 /**
  *  Apple Pay Button Component Props
@@ -50,13 +50,7 @@ export function ApplePayButton({
   const style = useMemo(() => mapButtonStyle(buttonStyle), [buttonStyle]);
 
   return (
-    <ApplePayButtonNative
-      type={buttonType}
-      buttonStyle={style}
-      borderRadius={borderRadius}
-      onPressAction={onPress}
-      {...props}
-    />
+    null
   );
 }
 
diff --git a/node_modules/@stripe/stripe-react-native/src/components/GooglePayButton.tsx b/node_modules/@stripe/stripe-react-native/src/components/GooglePayButton.tsx
index e8d97ad..d8b1039 100644
--- a/node_modules/@stripe/stripe-react-native/src/components/GooglePayButton.tsx
+++ b/node_modules/@stripe/stripe-react-native/src/components/GooglePayButton.tsx
@@ -8,7 +8,7 @@ import {
   StyleSheet,
 } from 'react-native';
 
-const GooglePayButtonNative = requireNativeComponent<any>('GooglePayButton');
+// const GooglePayButtonNative = requireNativeComponent<any>('GooglePayButton');
 
 /**
  *  Google Pay Button Component Props
@@ -48,7 +48,7 @@ export function GooglePayButton({
       onPress={onPress}
       style={disabled ? styles.disabled : {}}
     >
-      <GooglePayButtonNative buttonType={type} {...props} />
+      {/*<GooglePayButtonNative buttonType={type} {...props} />*/}
     </TouchableOpacity>
   );
 }

```

**Comment by charliecruzan-stripe** [comment]
Thanks for sharing that patch @walkerdb ! but if you are using rn < 64, you may run into this issue on Android: https://github.com/stripe/stripe-react-native/issues/919. Just wanted to note that and save you some googling

Let me know how the migration from `tipsi-stripe` and using the `Platform Pay` stuff works out, happy to chat anytime (my email is charliecruzan@stripe.com)

**Comment by walkerdb** [comment]
womp, thanks. `tipsi-stripe` was our react-native bump blocker, so I'll try updating to 0.64 to start and see what happens. Thanks for the heads up!

**Comment by walkerdb** [comment]
@charliecruzan-stripe hmm for the record, I still get the same error on react-native 0.64.4

**Comment by charliecruzan-stripe** [comment]
Let me take a look and see if i can reproduce that @walkerdb , thanks for all the info 🙏

**Comment by sanny-naked** [comment]
I was experiencing this error from simply wrapping a `View` with `StripeProvider`.

`"@stripe/stripe-react-native": "^0.22.0",`
`"react-native": "0.69.6",`

@walkerdb's patch has made it go away. Thank you.

**Comment by ravindra-encoresky** [comment]
I'm on  `react-native": 0.64.0` still I'm getting it.
I'm trying to migrate from tipsi-stripe.
the error is : **invariant Violation: Tried to register two views with the same name GooglePayButton**

**Comment by charliecruzan-stripe** [comment]
hey everyone- @walkerdb I was able to reproduce this using your `react-native.config.js` file. If you run your project without the `--config './react-native.config.js'`, do you get the error anymore?

**Comment by charliecruzan-stripe** [comment]
Fixed in Fixed in v0.22.1 ✅

**Comment by walkerdb** [comment]
confirmed thank you!!
